src/kronecker_generator.jl

```diff
@@ -1,10 +1,16 @@
-struct kroneckerState{T<:Integer, R<:Real}
-    edge::Array{T,2}
-    weight::Array{R}
+"""
+    Return the minimum Integer type which can contain `n`
+"""
+
+function get_min_type(n::Integer)
+  validtypes = [UInt8, UInt16, UInt32, UInt64]
+  for T in validtypes
+      n < typemax(T) && return T
+  end
 end
 
```

Comment:
Changed!
Also there should be an `<` and not `<=` because if `SCALE` is 8, the maximum index can be `256`.